agent/bench-scripts/pbench-linpack

```diff
@@ -228,9 +228,10 @@ for thread in ${threads//,/ }; do
 
 		for client in ${clients//,/ }; do
 			if pbench-is-local "${client}"; then
+				\rm -f "${iteration_dir}/linpack.finished"
```

Comment:
Why not run from the sample directory?

Then you don't have to remove `linpack.finished` each time.

Reply:
Is the leading back-slash supposed to be there?  (Ditto at line 234.)

It's really too bad that we cannot figure out how to put the command line in a variable, which would allow us to either run it locally or via `ssh`.  😖 Then we would only have to make these changes once....

Reply:
The problem is you need the `input` and copied `linpack` files and you produce some output files. In first iteration I always created everything fresh per each sample but I was asked to only generate this per iteration to avoid unnecessary distribution and copies of the same files.

Reply:
> I was asked to only generate this per iteration

That was me.  🥴

The question is, what can we do to minimize the amount of file generation and copying that we have to do.  Regenerating (substantially-)identical files seems like a waste; moving output files after they have been created is disappointing, unless there is no better approach; having to remember to remove the `.finished` file is, um, error-prone, apparently.  😉

So, if we could get the workload to produce the outputs in the directory which is supposed to receive them, that would be a help.  If we cannot do that, then perhaps we could run the workload in that directory and get it to pick up the input files from somewhere where they can be shared.  If the workload cannot be persuaded to do that, perhaps  we could get the effect we need with symlinks?  😛

Reply:
I wanted to avoid overengineering things, removing the file prior execution to ensure the expected file is not there at the time of beginning the script should work well in correct cases as well as in failed attempts (with a little chance of left-over process that finishes after the removal, but that should be a corner-case and shouldn't be happening). Adding an optional variable to store the `finished` file in a different location (or perhaps all output files) could be a bit safer but might introduce issues with the ansible-based execution (shouldn't, I know, but wanted to avoid modifying the file even further). Safest option would be to even expect a random hash in the `finished` file generated per each iteration, but IMO that would be an overkill.

Alternative would be to watch for the screen session to finish (which was actually my first implementation, but relying on `ls` of screen tmpfiles was odd).

Out of those I'd either keep the current solution or if we really want to be nice I'd go with the extra variable to overwrite the output location.

Reply:
As for the `\rm` I used that deliberately because I've already suffered systems with default `rm` alias to point to `rm -i` which stalled so I tend to use `\` to avoid using aliases.

Reply:
@webbnh @portante which direction would you like me to go? Should I add the output location suffix to the `linpack.sh` script, keep it the way it is or remove the lock files completely by reworking the way we execute the benchmark?

Reply:
I am currently studying the code to see what would be the best way forward.

Reply:
Posted PR https://github.com/ldoktor/pbench/pull/7.

If you feel like merging it great.  If like the changes, and they work for you, but your'd rather not usher this PR through to completion, just let me know and we can move these changes to a new PR which I can drive.  Whatever is easiest for you to manage.